pkg/pipelinescheduler/config_builder.go

```diff
@@ -6,10 +6,9 @@ import (
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
 	jenkinsv1 "github.com/jenkins-x/jx/v2/pkg/apis/jenkins.io/v1"
+	"github.com/jenkins-x/lighthouse-config/pkg/config"
```

Comment:
what is the advantage of extracting the lighthouse config into its own repo opposed to having a dedicated _api_ package in lighthouse itself which we could depend on?
I guess I am concerned about the amount of repository we are having to maintain. Do you think _lighthouse-config_ will ever have a different lifecycle than lighthouse itself?

Reply:
Versioning and avoiding circular dependencies. Since Lighthouse depends on jx for the metapipeline client, and extracting the metapipeline client from jx would be significantly tougher than extracting the config code from Lighthouse, this was the best way I could come up with to get jx to use Lighthouse’s config in the scheduler logic. The config will also release significantly less often than Lighthouse itself - as in I expect to make some changes in the next few weeks to add easier configurability of conditional execution, and then probably won’t be doing any releases in the foreseeable future, but if jx depends on Lighthouse, we’d need to either update jx’s dependency on Lighthouse every time Lighthouse releases (which in practice wouldn’t be an option since Lighthouse releases for each jx release!) or update the dependency manually, which would open up the possibility of forgetting to update jx when a relevant Lighthouse change was released.

I also just have an instinctive aversion to circular dependencies. =)

Reply:
Thanks for the explanation :-) I was not thinking about the circular dependency .